src/main/java/htsjdk/samtools/GenomicIndexUtil.java

```diff
@@ -52,6 +54,49 @@
      */
     public static final int UNSET_GENOMIC_LOCATION = 0;
 
+    /**
+     * Return the binning index level for the given bin
+     * @param bin
+     * @return the binning index level for the given bin
+     */
+    public static int bin2level(int bin) {
+        ValidationUtils.validateArg(bin >=0 && bin <= MAX_BINS, "Bin number must be >=0 and <= 37450");
+        // As described in Tabix: fast retrieval of sequence features from generic TAB-delimited files.
+        // doi: 10.1093/bioinformatics/btq671
+        return (int) Math.floor(((Math.log((7*bin) + 1) / Math.log(2)) / 3));
+    }
+
+    /**
+     * Return the binning index bin size for bins in the given bin
+     * @param level
+     * @return the bin size for bins at the given level
+     */
+    public static int level2Size(int level) {
+        ValidationUtils.validateArg(level >=0 && level <= 5, "Level number must be >=0 and <= 5");
+        // As described in Tabix: fast retrieval of sequence features from generic TAB-delimited files.
+        // doi: 10.1093/bioinformatics/btq671
+        return (int) (int) Math.pow(2, 29-(3*level));
+    }
+
+    /**
+     * Return a string describing the bin level, level size, and genomic territory covered by a bin.
+     * @param bin
+     * @return a string describing a bin
+     */
+    public static String bin2Region(int bin) {
```

Comment:
I'm not sure bin2Region makes it clear that this is producing debugging output.  Maybe "getDescriptionOfBin" or something like that.